src/Avalonia.Controls/Primitives/PopupPositioning/IPopupPositioner.cs

```diff
@@ -265,25 +265,25 @@ public static void ValidateGravity(this PopupGravity gravity)
 
         public static PopupAnchor Flip(this PopupAnchor edge)
         {
-            if (edge.HasFlagCustom(PopupAnchor.HorizontalMask))
+            if (edge.HasAnyFlag(PopupAnchor.HorizontalMask))
```

Comment:
Here and below are lines which cause the positioning problem in the `master` branch.